src/OceanSeaIceModels/InterfaceComputations/component_interfaces.jl

```diff
@@ -42,19 +43,65 @@ struct SeaIceOceanInterface{J, P, H, A}
     previous_ice_concentration :: A
 end
 
-struct ComponentInterfaces{AO, ASI, SIO, C, AP, OP, SIP, ATM}
+struct ComponentInterfaces{AO, ASI, SIO, C, AP, OP, SIP, EX}
     atmosphere_ocean_interface :: AO
     atmosphere_sea_ice_interface :: ASI
     sea_ice_ocean_interface :: SIO
     atmosphere_properties :: AP
     ocean_properties :: OP
     sea_ice_properties :: SIP
-    # Scratch space to hold the near-surface atmosphere state
-    # interpolated to the ocean grid
-    near_surface_atmosphere_state :: ATM
+    exchanger :: EX
     net_fluxes :: C
 end
 
+struct StateExchanger{G, AST, AEX}
+    exchange_grid :: G
+    exchange_atmosphere_state :: AST
+    atmosphere_exchanger :: AEX
+end
+
+function StateExchanger(ocean::Simulation, atmosphere)
+    # TODO: generalize this
+    exchange_grid = ocean.model.grid
+
+    exchange_atmosphere_state = (u  = Field{Center, Center, Nothing}(exchange_grid),
+                                 v  = Field{Center, Center, Nothing}(exchange_grid),
+                                 T  = Field{Center, Center, Nothing}(exchange_grid),
+                                 q  = Field{Center, Center, Nothing}(exchange_grid),
+                                 p  = Field{Center, Center, Nothing}(exchange_grid),
+                                 Qs = Field{Center, Center, Nothing}(exchange_grid),
+                                 Qℓ = Field{Center, Center, Nothing}(exchange_grid),
+                                 Mp = Field{Center, Center, Nothing}(exchange_grid))
+
+    # TODO: use kernel_parameters not hard code?
+    # kernel_parameters = interface_kernel_parameters(ocean_grid)
+    
+    atmos_grid = atmosphere.grid
+    arch = architecture(exchange_grid)
+    Nx, Ny, Nz = size(exchange_grid)
+
+    # Make an array of FractionalIndices
+    kᴺ = size(exchange_grid, 3)
+    X1 = _node(1, 1, kᴺ + 1, exchange_grid, c, c, f)
```

Comment:
will this work on a GPU?

Reply:
might need an @allowscalar

Reply:
the issue is for triply-flat grids. I could detect that case manually. But this seemed a bit more direct